modules/fedimint-mint-server/src/lib.rs

```diff
@@ -363,15 +361,143 @@ impl ServerModule for Mint {
+            let signature_shares = dbtx
+                .find_by_prefix(&ReceivedPartialSignatureKeyOutputPrefix(out_point))
+                .await
+                .map(|(key, partial_sig)| (key.1, partial_sig))
+                .collect::<Vec<_>>()
+                .await;
+
+            // check if we have enough signature shares to combine
+            if signature_shares.len() < self.cfg.consensus.peer_tbs_pks.threshold() {
+                continue;
+            }
+
+            // combine valid signature shares
+            let blind_signatures = TieredMultiZip::new(
+                signature_shares
+                    .iter()
+                    .map(|(_peer, sig_share)| sig_share.0.iter_items())
+                    .collect(),
+            )
+            .map(|(amt, sig_shares)| {
+                let peer_ids = signature_shares.iter().map(|(peer, _)| *peer);
+
+                let sig = combine_valid_shares(
+                    sig_shares
+                        .into_iter()
+                        .zip(peer_ids)
+                        .map(|((.., share), peer)| (peer.to_usize(), *share)),
+                    self.cfg.consensus.peer_tbs_pks.threshold(),
+                );
+
+                (amt, sig)
+            })
+            .collect::<TieredMulti<_>>();
+
+            // remove received signature shares
+            dbtx.remove_by_prefix(&ReceivedPartialSignatureKeyOutputPrefix(out_point))
+                .await;
+
+            // remove proposed signature share
+            dbtx.remove_entry(&ProposedPartialSignatureKey(out_point))
+                .await;
+
+            // insert the final blind signatures
+            dbtx.insert_entry(
+                &OutputOutcomeKey(out_point),
+                &MintOutputBlindSignatures(blind_signatures),
+            )
+            .await;
+
+            let mut redemptions = Amount::from_sats(0);
+            let mut issuances = Amount::from_sats(0);
+            let remove_audit_keys = dbtx
+                .find_by_prefix(&MintAuditItemKeyPrefix)
+                .await
+                .map(|(key, amount)| {
+                    match key {
+                        MintAuditItemKey::Issuance(_) => issuances += amount,
+                        MintAuditItemKey::IssuanceTotal => issuances += amount,
+                        MintAuditItemKey::Redemption(_) => redemptions += amount,
+                        MintAuditItemKey::RedemptionTotal => redemptions += amount,
+                    }
+                    key
+                })
+                .collect::<Vec<_>>()
+                .await;
+
+            for key in remove_audit_keys {
+                dbtx.remove_entry(&key).await;
+            }
+
+            dbtx.insert_entry(&MintAuditItemKey::IssuanceTotal, &issuances)
+                .await;
+            dbtx.insert_entry(&MintAuditItemKey::RedemptionTotal, &redemptions)
+                .await;
         }
```

Comment:
I think we need to find a better way to handle audits, doing them on every consensus item but *not* when transactions are processed seems too random. Previously this was done after every epoch and mechanical migrations of the code got us here. @jkitman do you have enough context to think about this? I could we move the DB compaction into the audit function of the module?

Reply:
To me it seems like it does not matter when the DB compaction runs (could even be different for every peer). Sensible options going forward seem to be either before every audit, hence after every transaction, or after every session - right now adding a function like on_session_switch to the ServerModule trait seems to make sense for other reasons anyways. However, for this pr we can keep it where it is and I can make a note to move it to a more sensible option later...